File: matrix_multiplicator.js
"use strict";

function fillArray(arr) {
    for (let i = 0; i < arr.length; i++) {
        arr[i] = i;
    }
    return arr;
}

function createMatrix(dimension){
    return fillArray(Array(dimension)).map(
        function(n) {
            return fillArray(new Float64Array(dimension))
        })
    }

function printMatrix(matrix){
    matrix.map((n) => {
        console.log(n);
    })
}

function multiplyMatrix(matrixA, matrixB) {
    let dim = matrixA.length;
    let matrixC = createMatrix(dim);
    
    for (let i = dim; --i >= 0;) {
        let matrix_c_i = matrixC[i];
        let matrixA_i = matrixA[i];
        for (let j = dim; --j >= 0;) {
            let matrix_c_i_j = 0;
            for (let k = dim; --k >= 0;) {
                matrix_c_i_j += matrixA_i[k] * matrixB[k][j];
            }
            matrix_c_i[j] = matrix_c_i_j;
        }
    }
    return matrixC;
}

for (let i = 100; i <= 2000; i += 100) {
    let matrixA = createMatrix(i);
    let matrixB = createMatrix(i);

    let start = Date.now();

    multiplyMatrix(matrixA, matrixB);

    let end = Date.now();

    console.log("Dim = " + i + '\n' + ((end - start)/1000) + ' s' + '\n');

}

File: MatrixMultiplicator.java
public class MatrixMultiplicator {

    public MatrixMultiplicator() {
    }

    public void printMatrix(float[][] matrixA) {
        int size = matrixA[0].length;
        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size; j++) {
                System.out.print("[" + matrixA[i][j] + "]");
            }
            System.out.println("");
        }
    }

    public static float[][] multiplicate(float[][] matrixA, float[][] matrixB) {
        int size = matrixA[0].length;
        float[][] matrixC = new float[size][size];
        for (int i = size; --i >= 0;) {
            float[] matrixA_i = matrixA[i];
            float[] matrixC_i = matrixC[i];
            for (int j = size; --j >= 0;) {
                float matrixC_i_j = 0;
                for (int k = size; --k >= 0;) {
                    matrixC_i_j += matrixA[i][k] * matrixB[k][j];
                }
                matrixC_i[j] = matrixC_i_j;
            }
        }
        return matrixC;
    }

    public static void main(String[] args) {
        float[][] matrixA, matrixB;
        long start;
        double elapsedTime;

        for (int n = 100; n <= 2000; n = n + 100) {
            matrixA = new float[n][n];
            matrixB = new float[n][n];
            for (int j = 0; j < n; j++) {
                for (int k = 0; k < n; k++) {
                    matrixA[j][k] = (float) Math.random();
                    matrixB[j][k] = (float) Math.random();
                }
            }
            start = System.currentTimeMillis();
            multiplicate(matrixA, matrixB);
            elapsedTime = ((double)(System.currentTimeMillis() - start))/1000;
            System.out.println("Dim = " + n + " Time: " + elapsedTime + " s");
        }
    }

}
